analyzers/src/SonarAnalyzer.Utilities/RuleDetailBuilder.cs

```diff
@@ -43,7 +43,7 @@ public static IEnumerable<RuleDetail> GetAllRuleDetails(AnalyzerLanguage languag
                 LanguageNames.VisualBasic => new ResourceManager("SonarAnalyzer.RspecStrings", typeof(Rules.VisualBasic.FlagsEnumZeroMember).Assembly),
                 _ => throw new InvalidOperationException("Unexpected language: " + language)
             };
-            return new RuleFinder()
+            return RuleFinder
                 .GetAnalyzerTypes(language)
                 .Select(x => (DiagnosticAnalyzer)Activator.CreateInstance(x))
```

Comment:
This may fail at runtime, because GetAnalyzerTypes does not filter our abstract classes.

Reply:
I've added another UT especially for that, as you've caught the `IsAbstract` issue above.